### PR TITLE
feat: `avm/res/container-service/managed-cluster` Update to new aks api version.

### PR DESCRIPTION
**Pull request by PixelRobots**
## Description

Updated Api version and fixed issues with the monitoring section to work with the new API.

Also fixed some part in the tests as a paramater was in the wrong place.

There also seems to be an issue in the Azure tets bicep due to timing's of managed identity and rbac on key vault so i am unable to get a fully green workflow run, but i have tested the code manually and it works.

## Pipeline Reference

| Pipeline |
| -------- |
|     [avm.res.container-service.managed-cluster](https://github.com/PixelRobots/bicep-registry-modules/actions/workflows/avm.res.container-service.managed-cluster.yml)     |

## Type of Change

- [ ] Update to CI Environment or utlities (Non-module effecting changes)
- [X] Azure Verified Module updates:
  - [ ] Bugfix containing backwards compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

## Checklist

- [X] I'm sure there are no other open Pull Requests for the same update/change
- [X] I have run `Set-AVMModule` locally to generate the supporting module files.
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

**Comment by PixelRobots** [comment]
Managed to get a green tick. took about 6 attempts to run and it finally worked. The tests really could do with some improvements.

**Comment by ChrisSidebotham** [comment]
Thankks for the Contribution @PixelRobots. Please do feedback here the issues you had with the tests and @ilhaan & @JPEasier can review this for improvments in the next versions.

@ilhaan @JPEasier - Please review this PR

**Review by ilhaan — DISMISSED** [review]

**Comment by AlexanderSehr** [comment]
@ilhaan, before the PR is merged, I just like to quickly call out that the pipeline is currently failing. In other words - even with the PR being merged there is a high chance the original issue will again fail the pipeline and the new version would not be published.

I commented on the 'Failed Pipeline' issue too:  https://github.com/Azure/bicep-registry-modules/issues/1966#issuecomment-2143807734

**Review by ChrisSidebotham — DISMISSED** [review]

File: avm/res/container-service/managed-cluster/main.bicep
Comment (eriqua) [inline avm/res/container-service/managed-cluster/main.bicep:1]: I'd suggest blocking this PR until discussion in #1966 is addressed https://github.com/Azure/bicep-registry-modules/issues/1966#issuecomment-2146193780

Comment (PixelRobots) [reply avm/res/container-service/managed-cluster/main.bicep:1]: I have removed auto instrumentation from my branch. The error in #1966 was before my change and seems to be an issue azure side as I do not have that preview feature enabled and it works. I had a customer the other day try to use the avm module and it failed with the same issue, another customer with the same settings and module it worked. (both do not have the preview feature registered). The only difference was the region I believe.

Comment (eriqua) [reply avm/res/container-service/managed-cluster/main.bicep:1]: Hey @PixelRobots yes, the pipeline failure is prior this contribution, and thus blocking yours. With a failed pipeline your update will not be published, so I'd suggest to fix the pipeline first.
Let's continue the conversation in the failed pipeline issue itself.

Comment (JPEasier) [reply avm/res/container-service/managed-cluster/main.bicep:1]: @eriqua I tested this changes in my own test environment, and it will fix the current problems in the main branch.

**Review by JPEasier — CHANGES_REQUESTED** [review]
Thanks a lot @PixelRobots for your great work!

only one little request from my side, please update also the Api version in the sub module `agent pool`.

**Comment by AlexanderSehr** [comment]
Hey @JPEasier, just added the `Needs: Moduke Owner' tag again as it's just something we use in BRM to differentiate whether a review is on an owner, or the maintainers :)

**Comment by PixelRobots** [comment]
> Thanks a lot @PixelRobots for your great work!
> 
> only one little request from my side, please update also the Api version in the sub module `agent pool`.

Updated and pushed changes. hopefully ok now.

**Review by JPEasier — APPROVED** [review]

**Pull request merged**
